addons/stock_account/models/stock_history.py

```diff
@@ -0,0 +1,156 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, tools
+from odoo.fields import Datetime
+
+
+class StockHistory(models.Model):
+    _name = 'stock.history'
+    _auto = False
+    _order = 'operation_date'
+
+    move_id = fields.Many2one('stock.move', 'Stock Move', required=True)
+    location_id = fields.Many2one('stock.location', 'Location', required=True)
+    company_id = fields.Many2one('res.company', 'Company')
+    product_id = fields.Many2one('product.product', 'Product', required=True)
+    product_categ_id = fields.Many2one('product.category', 'Product Category', required=True)
+    quantity = fields.Float('Product Quantity')
+    operation_date = fields.Datetime(oldname='date')
+    price_unit_on_quant = fields.Float('Value')
+    inventory_value=  fields.Float(compute='_get_inventory_value', string="Inventory Value", readonly=True)
+    source = fields.Char()
+    product_template_id = fields.Many2one('product.template', 'Product Template', required=True)
+    serial_number = fields.Char('Serial Number', required=True)
+
+    @api.model
+    def read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=False, lazy=True):
+        res = super(StockHistory, self).read_group(domain, fields, groupby, offset=offset, limit=limit, orderby=orderby, lazy=lazy)
+        Stock_History = self.env['stock.history']
```

Comment:
when you assign value of object then variable name should be camel case not snake case like : 
```
StockHistory = self.env['stock.history']
```